packages/runtime/runtime-utils/src/summaryUtils.ts

```diff
@@ -228,15 +234,29 @@ export class SummaryTreeBuilder implements ISummaryTreeWithStats {
 		this.summaryStats.handleNodeCount++;
 	}
 
+	/**
+	 * Adds a child and updates the stats accordingly.
+	 * @param key - The key to store the handle at in the current summary tree being generated. Should not contain any "/" characters.
+	 * The key should be unique within the current summary tree, and not transform when encodeURIComponent is called.
+	 * @param summarizeResult - Similar to {@link @fluidframework/runtime-definitions#ISummaryTreeWithStats}. The provided summary can be either a {@link @fluidframework/driver-definitions#ISummaryHandle} or {@link @fluidframework/driver-definitions#ISummaryTree}.
+	 */
 	public addWithStats(key: string, summarizeResult: ISummarizeResult): void {
 		this.summaryTree[key] = summarizeResult.summary;
 		this.summaryStats = mergeStats(this.summaryStats, summarizeResult.stats);
 	}
 
+	/**
+	 * Adds an attachment to the summary. This blob needs to already be uploaded to storage.
```

Comment:
Nothing in this doc comment or signature explains why "blob" is mentioned here. Or what an attachment is. Or how you get its id.

Linking to `ISummaryAttachment` or even 
Linking to `ISummaryAttachment.id` would probably help. Also if this has nothing to do with ISummaryBlob, avoiding the term blob would prabably be good.


Also given that ISummaryAttachment.id is undocumented, adding a doc to it and linking it would be good.

Note that ISummaryAttachment.id is read/write, so someone not knowing better could try and set the id to what ever they want, which I assume wouldn't work. Making its docs clear about that would be good (maybe fix the field to be readonly: its kinda late for that so if changes it would gets its own PR and go through API review).

Also seems odd that when adding blobs and handles you provide the key under which the blob will be stored, but when adding an attachment, it puts in under a key it generates itself in an undocumented way.

It seems like a really odd choice for this method to pick the key for you, and the rest to require the key.

If I want to store some attachment as part of my summary, how would the loading code read it back from the summary without knowing the key? This seems hard to use and inconsistent with the other APIs. If there is a reason it works this way, the docs should probably note it.

Reply:
Regardless of these issues, this PR is a clear improvement, so I'll approve.

Reply:
>Also seems odd that when adding blobs and handles you provide the key under which the blob will be stored, but when adding an attachment, it puts in under a key it generates itself in an undocumented way.
I don't know enough historical context to document this part, so I'm going to leave it as is here.